src/zui/ZUIPersonSelect.tsx

```diff
@@ -200,12 +200,34 @@ const MUIOnlyPersonSelect: FunctionComponent<ZUIPersonSelectProps> = (
 
   const [createPersonOpen, setCreatePersonOpen] = useState(false);
 
+  const [showAddPersonButton, setShowAddPersonButton] = useState(false);
+  const [wasLoading, setWasLoading] = useState(false);
+
+  useEffect(() => {
+    if (autoCompleteProps.isLoading) {
+      // If the autocomplete is loading, we set the wasLoading state to true
+      // This state allows to keep track whether a search in the user database has already happened
+      setWasLoading(true);
+    } else if (wasLoading && !autoCompleteProps.isLoading) {
+      // Finally if the autocomplete is not loading anymore and the wasLoading state is true
+      // we can show the add person button, since the search in the user database has already happened
+      setShowAddPersonButton(true);
+      setWasLoading(false);
+    }
```

Comment:
If I'm not mistaken, `wasLoading` is synchronized with `autoCompleteProps.isLoading`, by which I mean to say that whenever `isLoading` becomes truthy, `wasLoading` is set to `true`, and whenever `isLoading` becomes falsy, `wasLoading` is set to `false`.

The name `wasLoading` to me implies that it should remain true after the loading completes. Right now, it is being reset to false after loading finishes, and instead `showAddPersonButton` becomes true.

This makes me wonder whether two states are really necessary, or whether one could be derived from the other. If `wasLoading? could be set to true when loading happens, and only reset once a person is selected, then I would expect `showAddPersonButton` to not be necessary. Am I missing something?

Reply:
i finally had the time to look into this and yes you are correct. 
Can you have a look again? 😅